src/android/TalsecPlugin.kt

```diff
@@ -88,6 +93,24 @@ class TalsecPlugin : CordovaPlugin() {
     return true
   }
 
+  /**
+   * Method retrieves app icon for the given parameter
+   * @param packageName package name of the app we want to retrieve icon for
+   * @return PNG with app icon encoded as a base64 string
+   */
+  fun getAppIcon(callbackContext: CallbackContext?, args: JSONArray?): Boolean {
+    val packageName = args?.optString(0, null) ?: run {
+      callbackContext?.error("Missing packageName parameter in Talsec Native Plugin")
+      return false
+    }
+    // Perform the app icon encoding on a background thread
+    backgroundHandler.post {
+      val encodedData = Utils.getAppIconAsBase64String(cordova.context, packageName)
+      mainHandler.post { callbackContext?.success(encodedData) }
+    }
+    return true
```

Comment:
toto funkce se může chovat takto asynchronně? Že dáš true, že skončila a pak někdy dojde ještě ten callback?

Reply:
jo, JS reaguje na eventy z callbackContextu, návratovú hodnotu ten funkcie nedostane. Rovnako aj handler prežije túto metodu, pretože zostane stále zaregistrovaný